metta/setup/components/notebook.py

```
@@ -0,0 +1,255 @@
+import json
+import subprocess
+from pathlib import Path
+
+from metta.setup.components.base import SetupModule
+from metta.setup.registry import register_module
+from metta.setup.utils import error, success, warning
+
+
+@register_module
+class NotebookSetup(SetupModule):
+    """
+    Jupyter notebook support component.
+
+    This component handles installation and registration of the Metta Jupyter kernel.
+    It uses Jupyter's built-in discovery mechanism to find kernels, making it
+    cross-platform compatible (works on macOS, Linux, Windows).
+
+    The kernel is registered as 'metta-venv-kernel' with display name 'metta-venv-kernel',
+    allowing users to select it in Jupyter notebooks and labs.
+    """
+
+    # Kernel configuration constants
+    KERNEL_NAME = "metta-venv-kernel"
+    DISPLAY_NAME = "metta-venv-kernel"
+
+    @property
+    def description(self) -> str:
+        return "Jupyter notebook support with kernel registration"
+
+    def is_applicable(self) -> bool:
+        # This component is always applicable - all users need notebook support
+        return True
+
+    def _get_kernel_path(self, kernel_name: str) -> Path | None:
+        """
+        Get kernel path using Jupyter's discovery mechanism.
+
+        This method uses 'jupyter kernelspec list --json' to find kernels,
+        which is cross-platform and handles different OS kernel locations automatically.
+
+        Args:
+            kernel_name: Name of the kernel to find (e.g., 'metta-notebook-kernel')
+
+        Returns:
+            Path to kernel directory if found, None otherwise
+        """
+        try:
```

we want errors to propagate and not fail silently